OrbitLinuxTracing/GpuTracepointEventProcessorTest.cpp

```diff
@@ -0,0 +1,589 @@
+      ((timeline.length() + 1) << 16) | sizeof(dma_fence_signaled_tracepoint);
+  memcpy(event.tracepoint_data.get() + sizeof(dma_fence_signaled_tracepoint), timeline.c_str(),
+         timeline.length() + 1);
+  EXPECT_EQ(event.ExtractTimelineString(), timeline);
+  return event;
+}
+
+orbit_grpc_protos::GpuJob MakeGpuJob(int32_t tid, uint32_t context, uint32_t seqno,
+                                     std::string timeline, int32_t depth,
+                                     uint64_t amdgpu_cs_ioctl_time_ns,
+                                     uint64_t amdgpu_sched_run_job_time_ns,
+                                     uint64_t gpu_hardware_start_time_ns,
+                                     uint64_t dma_fence_signaled_time_ns) {
+  orbit_grpc_protos::GpuJob expected_gpu_job;
+  expected_gpu_job.set_tid(tid);
+  expected_gpu_job.set_context(context);
+  expected_gpu_job.set_seqno(seqno);
+  expected_gpu_job.set_timeline(std::move(timeline));
+  expected_gpu_job.set_depth(depth);
+  expected_gpu_job.set_amdgpu_cs_ioctl_time_ns(amdgpu_cs_ioctl_time_ns);
+  expected_gpu_job.set_amdgpu_sched_run_job_time_ns(amdgpu_sched_run_job_time_ns);
+  expected_gpu_job.set_gpu_hardware_start_time_ns(gpu_hardware_start_time_ns);
+  expected_gpu_job.set_dma_fence_signaled_time_ns(dma_fence_signaled_time_ns);
+  return expected_gpu_job;
+}
+
+void ExpectGpuJobEq(const orbit_grpc_protos::GpuJob& actual,
+                    const orbit_grpc_protos::GpuJob& expected) {
+  EXPECT_EQ(actual.tid(), expected.tid());
+  EXPECT_EQ(actual.context(), expected.context());
+  EXPECT_EQ(actual.seqno(), expected.seqno());
+  EXPECT_EQ(actual.timeline(), expected.timeline());
+  EXPECT_EQ(actual.depth(), expected.depth());
+  EXPECT_EQ(actual.amdgpu_cs_ioctl_time_ns(), expected.amdgpu_cs_ioctl_time_ns());
+  EXPECT_EQ(actual.amdgpu_sched_run_job_time_ns(), expected.amdgpu_sched_run_job_time_ns());
+  EXPECT_EQ(actual.gpu_hardware_start_time_ns(), expected.gpu_hardware_start_time_ns());
+  EXPECT_EQ(actual.dma_fence_signaled_time_ns(), expected.dma_fence_signaled_time_ns());
+}
+
+}  // namespace
+
+TEST(GpuTracepointEventProcessor, NeedsListener) {
+  static constexpr pid_t kTid = 42;
+  static constexpr uint32_t kContext = 1;
+  static constexpr uint32_t kSeqno = 10;
+  static const std::string kTimeline = "timeline";
+  static constexpr uint64_t kTimestampA = 100;
+  static constexpr uint64_t kTimestampB = 200;
+  static constexpr uint64_t kTimestampD = 300;
+
+  GpuTracepointEventProcessor processor;
+  processor.PushEvent(
+      MakeFakeAmdgpuCsIoctlPerfEvent(kTid, kTimestampA, kContext, kSeqno, kTimeline));
+  processor.PushEvent(MakeFakeAmdgpuSchedRunJobPerfEvent(kTimestampB, kContext, kSeqno, kTimeline));
+  EXPECT_DEATH(processor.PushEvent(
+                   MakeFakeDmaFenceSignaledPerfEvent(kTimestampD, kContext, kSeqno, kTimeline)),
+               "listener_ != nullptr");
+}
+
+TEST_F(GpuTracepointEventProcessorTest, OneJob) {
```

Comment:
How about "WillNotifyListenerWithOneGpuJob" or something more descriptive?

Reply:
I want to avoid mentioning "notifying the listener" in every name as it doesn't really add any information, that's just the basic of how we test the class, using whether the listener is notified or not. Similarly for "gpu job created", what does it add? I'll try to expand the names a bit where applicable.

Reply:
Sounds good. In general, I was just missing the "what to expect" in the names.